src/Orleans/Streams/Core/StreamSubscriptionHandle.cs

```diff
@@ -13,6 +13,7 @@ public abstract class StreamSubscriptionHandle<T> : IEquatable<StreamSubscriptio
     {
         public abstract IStreamIdentity StreamIdentity { get; }
 
+        public string ProviderName;
         /// <summary>
```

Comment:
yes, it will - if you want to be able to deserialize old stream subscription handles correctly, then you need to find some place which is unused in the old version and use that to indicate that this is a newer version.

Abstract classed can never be serialized, so you need to look an implementation. Eg: `isRewindable` is a `bool`, so it will be serialized as a `byte` (`SerializationTokenType.True` or `SerializationTokenType.False`). You can somehow use the unused bits of that byte (probably by changing it to a byte-backed enum). I'm not suggesting it's a good option, but it's an option.